src/silo/config/preprocessing_config.cpp

```diff
@@ -14,13 +14,21 @@ void PreprocessingConfig::validate() const {
    if (!std::filesystem::exists(input_directory)) {
       throw preprocessing::PreprocessingException(input_directory.string() + " does not exist");
    }
-   if (ndjson_input_filename.has_value() && metadata_file) {
+   if (ndjson_input_filename.has_value() && metadata_file.has_value()) {
       throw preprocessing::PreprocessingException(fmt::format(
          "Cannot specify both a ndjsonInputFilename ('{}') and metadataFilename('{}').",
          ndjson_input_filename.value().string(),
          metadata_file.value().string()
       ));
    }
+   if (!ndjson_input_filename.has_value() && !metadata_file.has_value()) {
+      throw preprocessing::PreprocessingException(fmt::format(
+         "Neither a ndjsonInputFilename ('{}') nor a metadataFilename('{}') was specified in the "
+         "preprocessing config or via the --metadataFilename / --ndjsonInputFilename option.",
+         ndjson_input_filename.value().string(),
+         metadata_file.value().string()
+      ));
```

Comment:
There is another access to undefined options in there ;) 

I also would not mention how to do it, since it might change (and the current version forgot environment variables). Users should simply consult the documentation (maybe that's worth adding to the error message).

```suggestion
      throw preprocessing::PreprocessingException(fmt::format(
         "Neither a {} nor a {} was specified.",
         METADATA_FILENAME_OPTION.toString(),
         NDJSON_INPUT_FILENAME_OPTION.toString()
      ));
```